Src/DependencyCollector/Shared/Implementation/TelemetryDiagnosticSourceListener.cs

```diff
@@ -0,0 +1,293 @@
+            string dbStatement = null;
+            string httpMethodWithSpace = null;
+            string httpUrl = null;
+            string peerAddress = null;
+
+            foreach (KeyValuePair<string, string> tag in currentActivity.Tags)
+            {
+                // interpret Tags as defined by OpenTracing conventions
+                // https://github.com/opentracing/specification/blob/master/semantic_conventions.md
+                switch (tag.Key)
+                {
+                    case "operation.name": // not defined by OpenTracing
+                        {
+                            telemetry.Name = tag.Value;
+                            continue; // skip Properties
+                        }
+                    case "operation.data": // not defined by OpenTracing
+                        {
+                            telemetry.Data = tag.Value;
+                            continue; // skip Properties
+                        }
+                    case "db.statement":
+                        {
+                            dbStatement = tag.Value;
+                            break;
+                        }
+                    case "error":
+                        {
+                            bool failed;
+                            if (bool.TryParse(tag.Value, out failed))
+                            {
+                                telemetry.Success = !failed;
+                                continue; // skip Properties
+                            }
+                            break;
+                        }
+                    case "http.status_code":
+                        {
+                            telemetry.ResultCode = tag.Value;
+                            continue; // skip Properties
+                        }
+                    case "http.method":
+                        {
+                            httpMethodWithSpace = tag.Value + " ";
+                            break;
+                        }
+                    case "http.url":
+                        {
+                            httpUrl = tag.Value;
+                            if (Uri.TryCreate(tag.Value, UriKind.RelativeOrAbsolute, out requestUri))
+                            {
+                                continue; // skip Properties
+                            }
+                            break;
+                        }
+                    case "peer.address":
+                        {
+                            peerAddress = tag.Value;
+                            break;
+                        }
+                    case "peer.hostname":
+                        {
+                            telemetry.Target = tag.Value;
+                            continue; // skip Properties
+                        }
+                    case "peer.service":
+                        {
+                            telemetry.Type = tag.Value;
+                            continue; // skip Properties
```

Comment:
I think type should be taken from"component" tag (i.e. protocol)
e.g. we can have http call to the service AND grpc call to another endpoint of the same service.

Reply:
so "component" describes the (client) library, while in dependencies type is about the other (server) side of the call. Also there are multiple Redis libraries available - they would all show as different dependency types in that case

Reply:
well, it's all about guidance we give and how simple is to follow it. Whatever tag is used, we should tell lib owners to use very high-level name like 'redis' and not 'stack-exchange-redis-version-1.2.3'. 

Still, if there is a http call to docDb and also a tcp call to the same docDb, dependency type should be different, I don't think peer.service would allow this scenario.

Reply:
I think I will start writing down the guidance and there it should be easier to see if anything is redundant/missing/inconsistent